jupyterlab_gcedetails/src/components/synchronized_sliders.tsx

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import { stylesheet } from 'typestyle';
+import Grid from '@material-ui/core/Grid';
+import Slider from '@material-ui/core/Slider';
+import { TextInput } from 'gcp_jupyterlab_shared';
+import { HardwareConfiguration } from '../data';
+
+interface Props {
+  /* Assumes sorted list with memory and cpu set to 0 for first value */
+  values: HardwareConfiguration[];
+  onCpuChange: (value: number) => void;
+  onMemoryChange: (value: number) => void;
+}
+
+interface State {
+  memoryValue: number;
+  cpuValue: number;
+  memoryInputDisplayValue: string;
```

Comment:
Do you really need to store the string representations of your values for display? It seems like they could easily be inferred in your render method.

Reply:
If I remove the inputDisplayValue variables then the slider automatically moves to different positions while the user is still typing into the input field (jumps from 2 to 20 while the user is entering 20 for example). I thought it would be a better user experience if I only change the value of the slider once the user has finished entering input and the input field loses focus. What do you think?

Reply:
Ahh I see. So if you calculate the values in the `render` method, they're always going to change as the user changes. This feels similar to what we discussed this morning in that we should try to debounce the change events for the field rather than change state on each value change. After a sufficient amount of time has passed (maybe 1s?), we can consider the user's edit to be complete.

Reply:
I played around with implementing the debounced handler but doing so also debounces displaying the value the user has typed in the text field (user types in 25 but experiences a delay in seeing 25 in the input field for example). I think its still valuable to have a debounced handler instead of only updating the state.value variable on input blur, but I think we are still going to need the extra inputDisplayValue variables in the state because we want the value displayed in the text field to change instantaneously but delay the slider change slightly as the user types into the text field, what do you think?

Reply:
Ahh I see I see. Then it probably makes sense to do what you suggest here. I'll leave it to you.